### DISCUSSION TITLE
Translate Hy manual in other languages

### DISCUSSION THREAD

**Discussion by mjobuda** [hylang/hy#2158] [Ideas]
Is there an interest in translating and publishing the Hy manual in other languages then only English?
I could offer my time in taking care of the polish translation

**Comment by Kodiologist** [comment]
Ultimately translations would be cool to have, but they would be risky because it would be easy for the translations to fall out of sync with the English, and then readers of the translation would be unwittingly reading the wrong docs. I'd say you should only do this if you can commit to updating the translation quickly. Even if you can, it would be wisest to wait until Hy 1.0, because big documentation changes will probably happen first, so you'll have to redo a lot of translation.

You could try translating only the tutorial, at least as a first step, which would reduce the burden of updating considerably.

**Reply by mjobuda** [reply]
Here you are:
https://htmlpreview.github.io/?https://github.com/mjobuda/hy/blob/master/docs/_build/html/tutorial_pl.html

Source is here:
https://github.com/mjobuda/hy/blob/master/docs/tutorial_pl.rst

**Reply by Kodiologist** [reply]
Nice, thanks. Can you commit to keeping this up to date, such that the translation matches the original upon each release of Hy? If so, look into Read the Docs's multiple-language support (I know it exists, but I don't know how it works, especially for only partly translated documentation trees) and submit a pull request.

**Reply by mjobuda** [reply]
You mean to commit forever and ever until the end of the world? Of course! 
No problem, but I'll add that I'm the most unreliable person I know.

And I also got already a big dislike for Sphinx. Now I'm looking how internationalization works. And I'm absolutely disgusted!!
Just take a look at this:
https://docs.readthedocs.io/en/stable/guides/manage-translations.html
or
https://www.sphinx-doc.org/en/master/usage/advanced/intl.html
and then they want me to open up an account on transifex.com and want me to read this:
https://docs.transifex.com/integrations/sphinx-doc

I love open source and think that Hy is awesome,
but dealing with these monstrosities is a lot you ask for! 
I see already that dealing with this nonsense will take more time then doing the translation itself.
Just to integrate the file I already did translate in proper rst format one has to deal with all this bureaucratic ceremonies.
For the love of Lisps and opensource I will spend a few hours to deal with this, but if I won't succeed in integrating my translations I will give up and strongly suggest to use different software for documenting Hy,
Not this bloated XX. It looks a little bit as the whole point of this procedure is to sell transifex services.

Maybe somebody knows a shortcut to circumvent all of this? 

I just tried to change the language at the readthedocs.org website itself and failed!!!
If multilanguage support does not work properly on their own site then I see Schwarz for translations with this system!

**Reply by Kodiologist** [reply]
> You mean to commit forever and ever until the end of the world?

I was thinking more along the lines of a few releases. As magnificent as Hy is, I doubt it will live to see the end of the world.

It looks to me that the procedure uses gettext, the traditional and widely used tool for translating free software, and you won't have to worry much about anything specific to Sphinx. Transifex looks to be optional. I don't know if there are any real alternatives to gettext.

Sphinx has issues, but I don't see us getting rid of it, because then we'd have to reimplement autodoc.
